### DISCUSSION TITLE
OGC-API for INSPIRE WFS Maritime Units data

### DISCUSSION THREAD

**Discussion by mattenp** [ldproxy/ldproxy#1230] [Setup] [answered]
Hi,

I want to use our INSPIRE WFS (deegree) in ldproxy.

URL : [https://gdi.bsh.de/en/mapservice/Administrative-Units-Maritime-Units-WFS?SERVICE=WFS&REQUEST=GetCapabilities&VERSION=2.0.0](https://gdi.bsh.de/en/mapservice/Administrative-Units-Maritime-Units-WFS?SERVICE=WFS&REQUEST=GetCapabilities&VERSION=2.0.0)

Configuration (/store/entities/providers/MU_INSPIRE.log (replaced yml because yml can not be uploaded ): 
[MU_INSPIRE.log](https://github.com/user-attachments/files/16193281/MU_INSPIRE.log)

attached

MaritimeBoundary is working 

Output browser:
![maritimeboundary_output](https://github.com/user-attachments/assets/b5ec3393-70d9-4f24-82d9-5fbd573e3569)

but Baseline has some problems:

Output browser: 
![baseline_output](https://github.com/user-attachments/assets/80516d06-8431-4daa-9a22-7af6162524df)

Log:
[ldproxy.log](https://github.com/user-attachments/files/16193163/ldproxy.log)

I suppose after some validation tests (XML Spy, INSPIRE Reference Validator) the geometry should be valid.
Do you have an idea to solve the problem?

Best regards,

Matthias

**Comment by cportele** [comment, marked as answer]
Matthias, I had a quick look at the GML. The Baseline feature does not have a geometry, but it has embedded data types (or objects) which each has a geometry. There is no general mapping how to represent such a feature in GeoJSON, which only supports a single geometry. My guess is that such cases are simply not supported by the current, generic code for WFS providers. The current code will probably look for a geometry property as a top-level property of the feature.

**Reply by mattenp** [reply]
Hi Clemens,
thanks a lot.
Do you plan to support such nested geometries?
Is the INSPIRE data model too complex for OGC-API?

![mu_inspire_data_model](https://github.com/user-attachments/assets/b05578e9-e6cc-47fd-908f-a11aa871330c)

Do you think that we shouldn't use complex INSPIRE data for ldproxy or what is your experience?
Perhaps dealing with flat INPIRE data is the key to get sucess with OGC-API (PostgreSQL or gpkg)?

We have also gpkg but this is also not working: the columns are not found
[https://gdi.bsh.de/de/data/Verwaltungseinheiten-Seegrenzen.gpkg](https://gdi.bsh.de/de/data/Verwaltungseinheiten-Seegrenzen.gpkg)

Are there demo files for gpkg or PostgreSQL?


Best regards an a nice weekend,

Matthias

**Comment by cportele** [comment]
The problem is not the OGC API Standards, they support complex and nested data structures, if these are needed.

The limitation is with the WFS Feature Provider, which is not actively maintained and always will have constraints due to the nature of WFS. We recommend to use SQL (PostGIS or GeoPackage) as the source of feature data. With an SQL feature provider you can support very complex data models, if you have to.

For some sample configurations have a look at https://github.com/ldproxy/demo. This includes the configurations, data and steps to set the APIs from https://demo.ldproxy.net up. 

As an aside, it is not straightforward to encode a Baseline feature in GeoJSON (or Shapefile, etc.), because of the geometries in BaselineSegment. This, however, is not an API issue, it is just that these feature formats cannot handle such constructs directly. What is probably needed is to upgrade the BaselineSegments to features. 

When I look into the GeoPackage, there are also no baseline segment geometries included, as far as I can tell after a quick look.

**Reply by mattenp** [reply]
Hi,

thanks a lot, we are now using PostgreSQL, works fine.

Best regards,

Matthias
